src/types/authenticated-request.ts

```diff
@@ -0,0 +1,42 @@
+export type AuthenticatedRequest = Request & {
```

Comment:
Interesting, my understanding was that each auth middleware defines it's own additions to context independently, e.g, there does not need to be a single union type that understand all the middleware like this.

Reply:
For this I used the same pattern thats used in seam-connect, I was having a lot of trouble with the types being passed around and inferred.